### PR TITLE
(vcredist*) Refactoring of vcredist2015 and vcredist2017

### PR DESCRIPTION
**Pull request by jberezanski**
As promised [here](https://github.com/chocolatey/chocolatey-coreteampackages/pull/420#issuecomment-263424410), I have crafted a new package for the "140" version of the Visual C++ Runtime, which is the version released with Visual Studio 2015 and subsequently updated in Visual Studio 2017.

The new package, `vcredist140`, contains all the installation and uninstallation logic, dependencies etc. The existing `vcredist2015` and `vcredist2017` packages are converted to meta-packages, pointing to appropriate versions of `vcredist140`.

The package logic has been rewritten from scratch and now supports the following features:
- detection of prerequisite Service Packs (Vista, 7),
- detection of this or higher version being already installed (to conserve bandwith of folks with metered connections),
- more robust uninstallation.

My plan is to publish to Chocolatey.org all versions of `vcredist140` corresponding to published versions of the 201x packages (i.e. 2015 RTM, 2015 Update 1/2/3-two versions, 2017 Preview5, RC1 and RC5), which is why I evolved this package starting from 2015 RTM. For the two `vcredist201x` packages I will publish one new version, which converts them to meta-packages and, in case of 2017, updates to RC5.

All package versions included in this PR are available for testing from [a MyGet feed](https://www.myget.org/F/jberezanski-coreteampackages-jbdev/api/v2).

I have also obtained permission from @pocki to move `vcredist2015` officially to this repo.

### CODE REVIEW

File: manual/vcredist140/vcredist140.nuspec [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <description>Microsoft Visual C++ Redistributable for Visual Studio 2017 RC installs run-time components of Visual C++ libraries. These components are required to run C++ applications that are developed using Visual Studio 2017 RC and link dynamically to Visual C++ libraries. The packages can be used to run such applications on a computer even if it does not have Visual Studio 2017 installed. These packages also install run-time components of C Runtime (CRT), Standard C++, MFC, C++ AMP, and OpenMP libraries.
+
+The Visual C++ Redistributable for Visual Studio 2017 RC consists of files vcruntime140.dll, msvcp140.dll, vcomp140.dll, vcamp140.dll, mfc140.dll and other.
+
+[Supported Operating Systems](https://www.visualstudio.com/productinfo/visual-studio-15-system-requirements-vs): Windows 10, Windows 8.1 / Server 2012 R2 (with KB2919355), Windows 8 / Windows Server 2012, Windows 7 SP1 (with KB3033929) / Server 2008 R2 SP1, Windows Vista SP2 / Server 2008 SP2, Windows XP SP3 / Windows Server 2003 SP2
+
+Note: on some systems, if KB2999226 is installed as a dependency of this package, the computer may need to be restarted before the installation of this package will succeed.
```
Comment (pascalberger) [inline manual/vcredist140/vcredist140.nuspec]: Use `##Notes` heading as per [§1.2.3](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#123-informative-description)

File: manual/vcredist2015/vcredist2015.nuspec [hunk 2]
```diff
@@ -3,25 +3,34 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
   <metadata>
     <id>vcredist2015</id>
-    <version>14.0.23506.0</version>
-    <title>Visual C++ Redistributable for Visual Studio 2015</title>
-    <authors>Microsoft</authors>
+    <version>14.0.24215.20170201</version>
+    <packageSourceUrl>https://github.com/chocolatey/chocolatey-coreteampackages/tree/master/manual/vcredist2015</packageSourceUrl>
     <owners>pocki_c,chocolatey</owners>
```
Comment (pascalberger) [inline manual/vcredist2015/vcredist2015.nuspec]: Add Chocolatey user first (see [§1.2.4](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#124-add-chocolatey-among-owners))

File: manual/vcredist2015/vcredist2015.nuspec [hunk 3]
```diff
@@ -3,25 +3,34 @@
...
 
-    The installation on Windows 8.1 or Windows Server 2012 R2 fails when you do not install the update KB2919355</description>
-    <summary>Run-time components that are required to run C++ applications that are built by using Visual Studio 2015</summary>
-    <releaseNotes>#### Program
-* Date published by Microsoft: 11/17/2015
-* Supported Operating Systems: Windows 10 , Windows 7 Service Pack 1, Windows 8, Windows 8.1, Windows Server 2003 Service Pack 2, Windows Server 2008 R2 SP1, Windows Server 2008 Service Pack 2, Windows Server 2012, Windows Vista Service Pack 2, Windows XP Service Pack 3
-* Known issues: http://go.microsoft.com/fwlink/?LinkId=691930
+[Supported Operating Systems](https://www.visualstudio.com/en-us/productinfo/vs2015-sysrequirements-vs): Windows 10, Windows 8.1 / Server 2012 R2 (with KB2919355), Windows 8 / Windows Server 2012, Windows 7 SP1 (with KB3033929) / Server 2008 R2 SP1, Windows Vista SP2 / Server 2008 SP2, Windows XP SP3 / Windows Server 2003 SP2
+
+Note: on some systems, if KB2999226 is installed as a dependency of this package, the computer may need to be restarted before the installation of this package will succeed.
```
Comment (pascalberger) [inline manual/vcredist2015/vcredist2015.nuspec]: Use `##Notes` heading as per [§1.2.3](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#123-informative-description)

File: manual/vcredist2017/vcredist2017.nuspec [hunk 4]
```diff
@@ -19,17 +19,16 @@
...
 
-**NOTE**: The installer replaces the existing installation of the Visual Studio 2015 Redistributable (i.e. performs an in-place upgrade).
-
-[Supported Operating Systems](https://www.visualstudio.com/productinfo/visual-studio-15-system-requirements-vs): Windows 10 / Server 2016, Windows 8.1 / Server 2012 R2 (with KB2919355), Windows Server 2012, Windows 7 SP1 (with KB3033929) / Server 2008 R2 SP1, Windows Vista SP2 / Server 2008 SP2, Windows Server 2003 SP2, Windows XP SP3
+[Supported Operating Systems](https://www.visualstudio.com/productinfo/visual-studio-15-system-requirements-vs): Windows 10, Windows 8.1 / Server 2012 R2 (with KB2919355), Windows 8 / Windows Server 2012, Windows 7 SP1 (with KB3033929) / Server 2008 R2 SP1, Windows Vista SP2 / Server 2008 SP2, Windows XP SP3 / Windows Server 2003 SP2
```
Comment (pascalberger) [inline manual/vcredist2017/vcredist2017.nuspec]: Use `##Notes` heading as per [§1.2.3](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#123-informative-description)

File: manual/vcredist2017/vcredist2017.nuspec [hunk 5]
```diff
@@ -3,12 +3,12 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
```
Comment (pascalberger) [inline manual/vcredist2017/vcredist2017.nuspec]: You might also want to add UTF-8 check here:
```
<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
```

File: manual/vcredist2015/vcredist2015.nuspec [hunk 6]
```diff
@@ -3,25 +3,34 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
```
Comment (pascalberger) [inline manual/vcredist2015/vcredist2015.nuspec]: You might also want to add UTF-8 check here:
```
<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
```

File: manual/vcredist140/vcredist140.nuspec [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    <title>Microsoft Visual C++ Redistributable for Visual Studio 2017 RC (2017-02-07)</title>
+    <authors>Microsoft</authors>
+    <projectUrl>https://blogs.msdn.microsoft.com/vcblog/2016/11/16/visual-studio-2017-rc-now-available/</projectUrl>
+    <iconUrl>https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/04e8f175a6f14a40854289817302bcada57de999/icons/vcredist140.png</iconUrl>
+    <copyright>http://www.microsoft.com/about/legal/en/us/IntellectualProperty/Copyright/Default.aspx</copyright>
+    <licenseUrl>http://msdn.microsoft.com/en-US/cc300389.aspx</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <docsUrl>https://msdn.microsoft.com/library/jj620919.aspx</docsUrl>
+    <bugTrackerUrl>https://visualstudio.uservoice.com/forums/121579-visual-studio</bugTrackerUrl>
+    <tags>Microsoft Visual C++ Redistributable 140 2017 admin</tags>
```
Comment (pascalberger) [inline manual/vcredist140/vcredist140.nuspec]: Tags in lower-case as per [$1.2.2](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#122-obligatory-tags)

File: manual/vcredist2017/vcredist2017.nuspec [hunk 8]
```diff
@@ -19,17 +19,16 @@
     <summary>Run-time components that are required to run C++ applications that are built by using Visual Studio 2017 RC</summary>
```
Comment (pascalberger) [inline manual/vcredist2017/vcredist2017.nuspec]: You might want to change tags to lower-case here too

**Comment by jberezanski** [comment]
> Please note that indentation is 2 spaces by the guidelines.

Oh? When did that happen?? I seem to recall the guidelines specifying 4 spaces earlier, why the change?
Personally, I don't mind it in nuspec, but I find code much more readable with 4 spaces.

I'll fix the nuspec issues, sure. As for the icon, I just reused KennethB's icon from the VisualStudio2015* packages. I suppose I can extract the icon from devenv.exe.

**Comment by pascalberger** [comment]
@jberezanski Yes, in the Wiki there was once written 4 (which also would be my personal preference :).

If I remember correctly @AdmiringWorm learnd me the lesson back then, so he might know more details why 2 and not 4.

**Comment by AdmiringWorm** [comment]
In all honesty, I don't really know why 2 instead of 4.
It was changed to 2 since since the guidelines and the `.editorconfig` was using different values, and since the `.editorconfig` file was added before the guidelines it was correct (IMO) to change the guidelines to follow the editorconfig file instead.

File: manual/vcredist2015/vcredist2015.nuspec (hunk 6 above)
Comment (jberezanski) [inline manual/vcredist2015/vcredist2015.nuspec]: The check is there already (line 2, not shown in this diff).

File: manual/vcredist2017/vcredist2017.nuspec (hunk 5 above)
Comment (jberezanski) [inline manual/vcredist2017/vcredist2017.nuspec]: Same as above, the check is there already (line 2, not shown in this diff).

**Comment by jberezanski** [comment]
I see. Well, I can live with that, I suppose.

All requested changes done.

**Comment by jberezanski** [comment]
Cool, thanks for reviewing!

**Pull request merged**